src/features/defi/providers/fox-eth-lp/hooks/useFoxEthLiquidityPool.ts

```diff
@@ -0,0 +1,268 @@
+          } else {
+            throw new Error('Bad hdwallet config')
+          }
+        })()
+
+        if (!broadcastTXID) {
+          throw new Error('Broadcast failed')
+        }
+        return broadcastTXID
+      } catch (error) {
+        console.warn(error)
+      }
+    },
+    [
+      connectedWalletEthAddress,
+      ethAsset.chainId,
+      ethAsset.precision,
+      supportedEvmChainIds,
+      uniswapRouterContract,
+      wallet,
+    ],
+  )
+
+  const removeLiquidity = useCallback(
+    async (foxAmount: string, ethAmount: string, lpAmount: string) => {
+      try {
+        if (!connectedWalletEthAddress || !uniswapRouterContract || !wallet) return
+        const chainAdapterManager = getChainAdapterManager()
+        const adapter = chainAdapterManager.get(ethAsset.chainId) as ChainAdapter<KnownChainIds>
+        if (!adapter)
+          throw new Error(`addLiquidityEth: no adapter available for ${ethAsset.chainId}`)
+        const value = bnOrZero(ethAmount)
+          .times(bnOrZero(10).exponentiatedBy(ethAsset.precision))
+          .toFixed(0)
+        const data = uniswapRouterContract?.interface.encodeFunctionData('removeLiquidityETH', [
+          FOX_TOKEN_CONTRACT_ADDRESS,
+          lpAmount,
+          calculateSlippageMargin(foxAmount),
+          calculateSlippageMargin(ethAmount),
+          connectedWalletEthAddress,
+          Date.now() + 1200000,
+        ])
+        const adapterType = adapter.getChainId()
+        const estimatedFees = await (adapter as unknown as EvmBaseAdapter<EvmChainId>).getFeeData({
+          to: UNISWAP_V2_ROUTER_ADDRESS,
+          value,
+          chainSpecific: {
+            contractData: data,
+            from: connectedWalletEthAddress,
+            contractAddress: FOX_TOKEN_CONTRACT_ADDRESS,
+          },
+        })
+        const result = await (async () => {
+          if (supportedEvmChainIds.includes(adapterType)) {
+            if (!supportsETH(wallet))
+              throw new Error(`addLiquidityEthFox: wallet does not support ethereum`)
+            const fees = estimatedFees.average as FeeData<EvmChainId>
+            const {
+              chainSpecific: { gasPrice, gasLimit, maxFeePerGas, maxPriorityFeePerGas },
+            } = fees
+            const shouldUseEIP1559Fees =
+              (await wallet.ethSupportsEIP1559()) &&
+              maxFeePerGas !== undefined &&
+              maxPriorityFeePerGas !== undefined
+            if (!shouldUseEIP1559Fees && gasPrice === undefined) {
+              throw new Error(`addLiquidityEthFox: missing gasPrice for non-EIP-1559 tx`)
+            }
+            return await (adapter as unknown as ethereum.ChainAdapter).buildCustomTx({
+              to: UNISWAP_V2_ROUTER_ADDRESS,
+              value,
+              wallet,
+              data,
+              gasLimit,
+              bip44Params: adapter.buildBIP44Params({
+                accountNumber: 0,
```

Comment:
Magic number, can you comment why this is 0?

Reply:
this is cancer throughout our stack and multi account support is a long ways off, this is ok for now IMO